cmake/external/variant.cmake

```diff
@@ -0,0 +1,17 @@
+SET(VARIANT_RELEASE_VERSION v1.3.0)
+SET(VARIANT_SOURCE_DIR ${THIRD_PARTY_DIR}/variant)
+SET(VARIANT_INCLUDE_DIR ${THIRD_PARTY_INCLUDE_DIR}/variant)
+include(ExternalProject)
+ExternalProject_Add(
+		variant
+		PREFIX ${CMAKE_BINARY_DIR}/variant
+		SOURCE_DIR ${VARIANT_SOURCE_DIR}
+		GIT_REPOSITORY https://github.com/mpark/variant.git
+		GIT_SHALLOW 1
+		GIT_TAG single-header
```

Comment:
this should be ${VARIANT_RELEASE_VERSION} no?

Reply:
this is a branch for release, each version is a single directory

Reply:
yeah but they have tags as well, see https://github.com/mpark/variant/tree/v1.3.0

Reply:
because unfortunately the branch HEAD will always contain the latest release, meaning you will always checkout the latest release which probably not going to be in future `VARIANT_RELEASE_VERSION` and it's a good idea in general to depend on a *specific* version of an external lib.

Reply:
install with provided cmakelist requires a higher version cmake than ours
https://github.com/mpark/variant/blob/29319715a1f0eb0980d380db8a2fda5af8d58feb/CMakeLists.txt#L8